wall_e/extensions/sfu.py

```diff
@@ -502,6 +507,120 @@ async def outline(self, ctx, *course):
         if e_obj is not False:
             await ctx.send(embed=e_obj, reference=ctx.message)
 
+    async def _embed_followup_error_message(self, interaction: discord.Interaction, title, desc):
+        e_obj = await embed(
+            self.logger, interaction=interaction,
+            title=title,
+            description=desc,
+            colour=WallEColour.ERROR,
+        )
+        if e_obj:
+            msg = await interaction.followup.send(embed=e_obj)
+            await asyncio.sleep(10)
+            await msg.delete()
+
+    def _parse_courses_to_embed(self, courses: list) -> list:
+        content_to_embed = []
+        number_of_courses_per_page = 20
+        number_of_courses = 0
+        content = ""
+
+        for i, course in enumerate(courses):
+            course_title = course["title"]
+            course_number = course["text"]
+            content += f"\n{course_number} - {course_title}"
+
+            number_of_courses += 1
+            if i >= 0 and (number_of_courses % number_of_courses_per_page == 0 or i == len(courses) - 1):
+                number_of_courses = 0
+                content_to_embed.append(
+                    [["Code - Title", content]]
+                )
+                content = ""
+
+        return content_to_embed
+
+    @app_commands.command(name="courses", description="Gets all offered courses")
+    @app_commands.describe(department="Specify the department to search. Examples: STAT, PHYS, MATH")
+    @app_commands.describe(level="Specify the level of courses to filter for. Examples: 100, 200, 400")
+    @app_commands.describe(term="Specify the semester to search for. Requires year to be specified. "
+                                "Examples: Spring, Summer, Fall")
+    @app_commands.describe(year="Specify the year to search for. Requires term to be specified. "
+                                "Examples: 2020, 2024, 2025")
+    async def courses(self, interaction: discord.Interaction, department: str = "", level: Optional[int] = None,
+                      term: str = "registration", year: str = "registration"):
+        self.logger.info(
+            f"[SFU courses()] courses command detected from user {interaction.user} with arguments: "
+            f"department {department}, level {level}, term {term}, year {year}"
+        )
+        await interaction.response.defer()
+        err_msg_title = "SFU Courses Error"
+
+        # The default course selection if not specified
+        departments = ["CMPT", "MATH", "MACM"]
+        if department:
+            departments = [department.upper()]
+
+        if level and (level < 100 or level >= 1000):
```

Comment:
Just a little thing when checking for None values, you'll want to explicitly use `x is None` or `x is not None`. See [here](https://stackoverflow.com/a/3965165/11909458) for more details (in our case: `level = 0`, `if level` evaluates to false)

Same with line 595

Reply:
https://github.com/CSSS/wall_e/pull/666/commits/776ca4b3463cf29305262178779d768334dd9853